### PR TITLE
LPS-173652 | Test Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-161638 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-161638](https://issues.liferay.com/browse/LPS-161638)  → [LPS-173652](https://issues.liferay.com/browse/LPS-173652)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/CategoriesFragment.testcase [hunk 1]
```diff
@@ -0,0 +1,1045 @@
...
+		}
+	}
+
+	@description = "This is a test for LPS-161638. The categories field can be used in a master page"
+	@priority = 3
+	test CanUseCategoriesFieldInMasterPage {
+		task ("Given a master page with the categories fragment") {
+			JSONCategory.addVocabulary(
+				groupName = "Global",
+				title = "Public Global Vocabulary");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/CategoriesFragment.testcase:37]: Use randomName for vocabulary

File: portal-web/test/functional/com/liferay/portalweb/macros/asset/categories/AssetCategorization.macro [hunk 2]
```diff
@@ -208,6 +208,14 @@ definition {
...
 	}
 
+	macro selectVocabularyType {
+		ClickNoError(locator1 = "AssetCategorization#VOCABULARY_SELECT");
+
+		Select(
+			locator1 = "AssetCategorization#VOCABULARY_SELECT_OPTION",
+			value1 = ${option});
+	}
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/asset/categories/AssetCategorization.macro:218]: Move to PageEditor.configureFragment

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/AssetCategorization.path [hunk 3]
```diff
@@ -45,6 +45,16 @@
...
+<tr>
+	<td>VOCABULARY_SELECT</td>
+	<td>//div[contains(.,'Vocabulary Type')]//select</td>
+	<td></td>
+</tr>
+<tr>
+	<td>VOCABULARY_SELECT_OPTION</td>
+	<td>//div[contains(@id,'formInputGeneral')]//select</td>
+	<td></td>
+</tr>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/AssetCategorization.path:57]: No need

File: portal-web/test/functional/com/liferay/portalweb/macros/asset/categories/AssetCategorization.macro [hunk 4]
```diff
@@ -208,6 +208,14 @@ definition {
 			value1 = ${tagName});
 	}
 
+	macro selectVocabularyType {
+		ClickNoError(locator1 = "AssetCategorization#VOCABULARY_SELECT");
+
+		Select(
+			locator1 = "AssetCategorization#VOCABULARY_SELECT_OPTION",
+			value1 = ${option});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/asset/categories/AssetCategorization.macro:216]: The type of field is select
Just Use
```
			Select(
				key_fieldLabel = "Vocabulary Type",
				locator1 = "Select#GENERIC_SELECT_FIELD",
				value1 = ${vocabularyType});
```

File: portal-web/test/functional/com/liferay/portalweb/macros/asset/categories/AssetCategorization.macro [hunk 5]
```diff
@@ -237,7 +245,9 @@ definition {
 	}
 
 	macro viewNoVocabularyField {
-		Panel.expandPanel(panel = "Categorization");
+		if ((IsElementPresent(locator1 = "Panel#PANEL_COLLAPSED")) && (IsVisible(locator1 = "Panel#PANEL_COLLAPSED"))) {
+			Panel.expandPanel(panel = "Categorization");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/asset/categories/AssetCategorization.macro:250]: These logical is already defined in Panel.expandPanel macro, isn't it?

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind refactor according to comments? Thanks

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind refactor according to comments? Thanks

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/AssetCategorization.path [hunk 6]
```diff
@@ -45,6 +45,11 @@
 	<td>//*[@class='field-content']//label[normalize-space(text())='${key_fieldLabel}']</td>
 	<td></td>
 </tr>
+<tr>
+	<td>VOCABULARY_SELECT_OPTION</td>
+	<td>//div[contains(@id,'formInputGeneral')]//select</td>
+	<td></td>
+</tr>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/AssetCategorization.path:52]: Use the general locator

**Comment by Tim-Cao** [comment]
@rafaellajordao Good job! See changes https://github.com/liferay-echo/liferay-portal/pull/12061

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**